libraries/core_libs/consensus/include/dag/sortition_params_manager.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "pbft/sync_block.hpp"
+#include "storage/storage.hpp"
+#include "vdf/config.hpp"
+
+namespace taraxa {
+
+struct SortitionParamsChange {
+  uint64_t period = 0;
+  VrfParams vrf_params;
+  uint16_t interval_efficiency = 0;
+  uint16_t actual_correction_per_percent = 1;
+
+  SortitionParamsChange() = default;
+  SortitionParamsChange(uint64_t period, uint16_t efficiency, VrfParams const& vrf);
+  SortitionParamsChange(uint64_t period, uint16_t efficiency, VrfParams const& vrf,
+                        SortitionParamsChange const& previous);
+  static SortitionParamsChange from_rlp(dev::RLP const& rlp);
+  bytes rlp() const;
+
+  /**
+   * Calculates needed correction of VRF params per DAG efficiency percent
+   * This calculates parameter change which is necessary to change the efficiency by 1 percent. For example:
+   * Current efficiency is 72%. Based on past calculations, we changed the parameter to 100.
+   * The interval efficiency after these changes was 74%. It turns out that it has changed by 2%, but not the 3% as we
+   * need Therefore, the required changePerPercent is equal `100 / 2 = 50`, and not `100 / 3 = 33.(3)` as it was before
+   */
+  uint16_t correctionPerPercent(SortitionParamsChange const& previous) const;
+};
+
+class SortitionParamsManager {
+  SortitionConfig config_;
```

Comment:
we are trying to follow google c++ style. Class should start with public section, followed by protected an private:
https://google.github.io/styleguide/cppguide.html#Declaration_Order